src/phpDocumentor/ApiReference/Factory.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.5
+ *
+ * @copyright 2010-2015 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\ApiReference;
+
+use InvalidArgumentException;
+use phpDocumentor\DocumentGroup;
+use phpDocumentor\DocumentGroupDefinition as DocumentGroupDefinitionInterface;
+use phpDocumentor\DocumentGroupFactory;
+use phpDocumentor\Reflection\DocBlockFactory;
+use phpDocumentor\Reflection\Php\Factory\File\FlySystemAdapter;
+use phpDocumentor\Reflection\Php\NodesFactory;
+use phpDocumentor\Reflection\Php\ProjectFactory;
+use phpDocumentor\Reflection\Php\Factory as ProjectFactoryStrategy;
+use phpDocumentor\Reflection\PrettyPrinter;
+
+final class Factory implements DocumentGroupFactory
+{
+    /**
+     * Creates Document group using the provided definition.
+     *
+     * @param DocumentGroupDefinitionInterface $definition
+     * @return DocumentGroup
+     */
+    public function create(DocumentGroupDefinitionInterface $definition)
+    {
+        /** @var DocumentGroupDefinition $definition */
+        if (!$this->matches($definition)) {
+            throw new InvalidArgumentException('Definition must be an instance of ' . DocumentGroupDefinition::class);
+        }
+
+        $projectFactory = new ProjectFactory(
```

Comment:
Can we inject this factory into this one? I would prefer it if every instantiation here is handled by the Dependency Injection Container. In the ContainerDefinitions file you can define the array with dependencies that you want to inject in the ProjectFactory's first argument